# Consulting Data Analysis Week #1

In [13]:
# Import Pandas & Read in Data

import pandas as pd

contracts = pd.read_csv("C:/Users/Michael Botelho/bip-python-unh/sales_analysis/data/contracts.csv", delimiter =',', encoding="utf-8-sig")
customers = pd.read_csv("C:/Users/Michael Botelho/bip-python-unh/sales_analysis/data/customers.csv", delimiter =',', encoding="utf-8-sig")


# Changing Data Types

customers['id'] = customers['id'].astype(str)
contracts['customer_id'] = contracts['customer_id'].astype(str)



# Merge/Join to one DataFrame

df = pd.merge(customers, contracts, left_on='id', right_on='customer_id')


# Fix Dollar Signs & Commas & Convert to Float

df['deposit'] = df['deposit'].replace('[$,()]', '', regex=True)
df['deposit'] = df['deposit'].astype(float)



# Calculate Total Sales By State

total_sales = df.groupby(['state'])['deposit'].agg('sum')
total_sales = df.groupby(['state'])['deposit'].agg('mean', 'sum')
pd.set_option('display.max_columns', None)
total_sales.columns = ['state', 'total sales']
print(total_sales)

state
AK    61125.000000
AL    72571.428571
AR    47000.000000
AZ    46909.090909
CA    52454.545455
CO    55558.823529
CT    76454.545455
DC    53000.000000
DE    48357.142857
FL    54557.142857
GA    52000.000000
ID    54722.222222
IL    58608.695652
IN    45909.090909
KS    52674.418605
KY    73818.181818
LA    55484.848485
MA    59333.333333
MD    54388.888889
ME    52750.000000
MI    59300.000000
MN    56166.666667
MO    62576.923077
MS    47166.666667
NC    53680.000000
NE    45636.363636
NV    61777.777778
NY    54465.116279
OH    54878.787879
OK    49029.411765
OR    52272.727273
PA    56655.172414
SC    63200.000000
TN    55125.000000
TX    57826.923077
VA    56142.857143
WA    52407.407407
WI    52416.666667
WV    57000.000000
Name: deposit, dtype: float64


# Consulting Data Analysis Week #2

### 1. Identify the number of unique companies you do business with in each state

In [2]:
import pandas as pd

unique_companies = df.drop_duplicates(subset=['company'])

pd.Series.nunique(unique_companies['company'])

162

### 2. Complete any other analysis on the data that helps you illustrate the best state to setup a headquarters to support your customers and office

Overall Dataframe with all values


In [3]:
print(df.value_counts('state'))

state
TX    104
CA     88
FL     70
WA     54
NY     43
KS     43
VA     35
OK     34
CO     34
OH     33
LA     33
DC     32
PA     29
MO     26
NC     25
WI     24
IL     23
OR     22
AZ     22
NE     22
MI     20
ID     18
MN     18
MD     18
TN     16
GA     15
DE     14
KY     11
IN     11
CT     11
NV      9
AK      8
AL      7
MA      6
MS      6
WV      5
SC      5
ME      4
AR      2
dtype: int64


Unique DataFrame with unique values for company

In [4]:
unique_companies.value_counts('state')

state
TX    15
CA    14
FL    14
NY    10
WA     7
VA     7
KS     6
PA     6
DC     6
WI     5
OK     5
AZ     5
IL     4
CO     4
OR     4
TN     4
LA     4
ID     3
GA     3
DE     3
IN     3
MD     3
MO     3
NC     3
OH     3
MI     3
MN     2
NE     2
NV     2
CT     2
AR     1
AL     1
WV     1
KY     1
MA     1
ME     1
AK     1
dtype: int64

### 3. Complete any other analysis on the data that helps you illustrate the best state to setup a headquarters to support your customers and office 

Average sales by state per contracts by state

In [17]:
avg_sales = df.groupby(['state'])['deposit'].agg('mean','sum')

avg_sales / df.value_counts('state')

state
AK     7640.625000
AL    10367.346939
AR    23500.000000
AZ     2132.231405
CA      596.074380
CO     1634.083045
CT     6950.413223
DC     1656.250000
DE     3454.081633
FL      779.387755
GA     3466.666667
ID     3040.123457
IL     2548.204159
IN     4173.553719
KS     1224.986479
KY     6710.743802
LA     1681.359045
MA     9888.888889
MD     3021.604938
ME    13187.500000
MI     2965.000000
MN     3120.370370
MO     2406.804734
MS     7861.111111
NC     2147.200000
NE     2074.380165
NV     6864.197531
NY     1266.630611
OH     1662.993572
OK     1442.041522
OR     2376.033058
PA     1953.626635
SC    12640.000000
TN     3445.312500
TX      556.028107
VA     1604.081633
WA      970.507545
WI     2184.027778
WV    11400.000000
dtype: float64

## Which State?

As we see through the results: Texas, California, Florida, and New York are the states where most companies are located. On the first original dataframe Washington has more values than New York, but on the unique dataframe we see that New York has 10 unique companies, while Washington has 7 unique companies. This likely occured because the companies in Washington have more rows or contracts and deposits compared to the companies in New York.

Overall, it seems that the best state to have a headquarters in would be Texas because they have the most companies within the state at 15 and they have the most total rows at 104, which means that Texas likely has the most total deposits and contracts as well.